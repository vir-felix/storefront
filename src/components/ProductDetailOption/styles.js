export default (theme) => ({
  variantOptionContainer: {
    display: "flex",
    width: `${theme.spacing.unit * 3.125}%`,
    maxWidth: `${theme.spacing.unit * 3.125}%`,
    padding: theme.spacing.unit * 1.25
  },
  optionButton: {
    "flex": "1 1 auto",
    "fontWeight": 600,
    "padding": theme.spacing.unit * 1.25,
    "borderRadius": "2px",
    "backgroundColor": theme.palette.primary.contrastText,
    "border": "1px solid",
    "borderColor": theme.palette.primary.borderColor,
    "&:hover": {
      backgroundColor: theme.palette.reaction.activeElementBackground
    }
  },
  optionText: {
    fontWeight: 500,
    fontSize: "14px"
  },
  isSelected: {
    backgroundColor: theme.palette.reaction.activeElementBackground
  }
});
