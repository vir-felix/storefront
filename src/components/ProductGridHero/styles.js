export default (theme) => ({
  heroImg: {
    width: "100%",
    height: "325px",
    objectFit: "cover"
  },
  productGridContainer: {
    maxWidth: theme.grid.productGridMaxWidth,
    marginLeft: "auto",
    marginRight: "auto"
  }
});
