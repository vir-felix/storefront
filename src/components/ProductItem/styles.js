export const styles = (theme) => ({
  "@keyframes spin": {
    from: { transform: "rotate(0deg)" },
    to: { transform: "rotate(-360deg)" }
  },
  "root": {},
  "productInfo": {
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    paddingTop: 10
  },
  "productMedia": {
    backgroundColor: theme.palette.primary.contrastText,
    position: "relative"
  },
  "img": {
    height: "auto",
    width: "100%"
  },
  "imgLoading": {
    alignItems: "center",
    backgroundColor: "transparent",
    display: "flex",
    fontSize: 48,
    height: 100,
    justifyContent: "center",
    left: "calc(50% - 50px)",
    position: "absolute",
    top: "calc(50% - 50px)",
    width: 100
  },
  "loadingIcon": {
    color: theme.palette.primary.light,
    fontSize: "inherit",
    animationName: "spin",
    animationDuration: theme.transitions.duration.standard * 2,
    animationTimingFunction: theme.transitions.easing.sharp,
    animationIterationCount: "infinite",
    animationFillMode: "both"
  },
  "link": {
    ...theme.typography.body2,
    "textAlign": "left",
    "&:hover": {
      color: theme.palette.secondary.main
    }
  }
});
