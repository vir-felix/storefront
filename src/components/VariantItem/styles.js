export default (theme) => ({
  variantButton: {
    "display": "flex",
    "justifyContent": "space-between",
    "textTransform": "none",
    "width": "100%",
    "borderRadius": "2px",
    "backgroundColor": theme.palette.primary.contrastText,
    "border": "1px solid",
    "borderColor": theme.palette.primary.contrastText,
    "paddingTop": theme.spacing.unit * 1.25,
    "paddingBottom": theme.spacing.unit * 1.25,
    "paddingLeft": theme.spacing.unit * 2.5,
    "paddingRight": theme.spacing.unit * 2.5,
    "&:hover": {
      border: "1px solid",
      borderColor: theme.palette.reaction.activeElementBorderColor,
      backgroundColor: theme.palette.primary.contrastText
    }
  },
  activeVariant: {
    border: "1px solid",
    borderColor: theme.palette.reaction.activeElementBorderColor
  }
});
