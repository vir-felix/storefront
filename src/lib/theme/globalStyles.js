export default
<style global jsx>
  {`
    a {
      color: inherit;
      text-decoration: none;
    }
  `}
</style>;
