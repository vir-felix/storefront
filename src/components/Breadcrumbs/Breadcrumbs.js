import React, { Component, Fragment } from "react";
import PropTypes from "prop-types";
import { inject, observer } from "mobx-react";
import { withStyles } from "@material-ui/core/styles";
import ChevronRight from "mdi-material-ui/ChevronRight";
import Link from "components/Link";

import withTags from "containers/tags/withTags";

const styles = (theme) => ({
  container: {
    display: "flex",
    alignItems: "center",
    maxWidth: theme.grid.productGridMaxWidth,
    marginLeft: "auto",
    marginRight: "auto"
  },
  breadcrumbLink: {
    fontSize: "14px",
    letterSpacing: "0.3px",
    color: "#3c3c3c",
    border: 0,
    borderBottom: "1px solid",
    borderColor: theme.palette.reaction.borderColor,
    margin: "0 7px"
  },
  breadcrumbIcon: {
    fontSize: "14px"
  }
});


@withStyles(styles)
class Breadcrumbs extends Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    isPDP: PropTypes.bool,
    isTagGrid: PropTypes.bool,
    product: PropTypes.object,
    tag: PropTypes.object,
    tags: PropTypes.object
  }

  get isPDP() {
    const { isPDP } = this.props;
    return isPDP || false;
  }

  get isTagGrid() {
    const { isTagGrid } = this.props;
    return isTagGrid || false;
  }

  get isTopLevel() {
    const { tag } = this.props;

    return tag && tag.isTopLevel;
  }

  renderTopLevelTagBreadcrumbs = (tag) => {
    const { classes: { breadcrumbIcon, breadcrumbLink }, tag: propTag } = this.props;

    const currentTag = tag || propTag;

    return (
      <Fragment>
        <ChevronRight className={breadcrumbIcon} />
        <Link route={`/tag/${currentTag.slug}`}><span className={breadcrumbLink}>{currentTag.name}</span></Link>
      </Fragment>
    );
  }

  renderSecondLevelTagBreadcrumbs = (tag) => {
    const { classes: { breadcrumbIcon, breadcrumbLink }, tag: propTag, tags } = this.props;
    const currentTag = tag || propTag;

    // Find tag that is a parent of this tag
    const nodes = tags.edges.map((edge) => edge.node);
    const parentTag = nodes.find((node) => node.subTagIds.includes(currentTag._id));

    return (
      <Fragment>
        {this.renderTopLevelTagBreadcrumbs(parentTag)}
        <ChevronRight className={breadcrumbIcon} />
        <Link route={`/tag/${currentTag.slug}`}><span className={breadcrumbLink}>{currentTag.name}</span></Link>
      </Fragment>
    );
  }

  renderTagGridBreadcrumbs = (tag) => {
    if (this.isTopLevel || (tag && tag.isTopLevel)) {
      return this.renderTopLevelTagBreadcrumbs(tag);
    }
    return this.renderSecondLevelTagBreadcrumbs(tag);
  }

  renderPDPBreadCrumbs = () => {
    const { classes: { breadcrumbIcon, breadcrumbLink }, product, tag } = this.props;

    if (tag && tag._id) {
      return (
        <Fragment>
          {this.renderTagGridBreadcrumbs(tag)}
          <ChevronRight className={breadcrumbIcon} />
          <Link route={`/product/${product.slug}`}><span className={breadcrumbLink}>{product.title}</span></Link>
        </Fragment>
      );
    }

    return (
      <Fragment>
        <ChevronRight className={breadcrumbIcon} />
        <Link route={`/product/${product.slug}`}><span className={breadcrumbLink}>{product.title}</span></Link>
      </Fragment>
    );
  }

  renderBreadcrumbs = () => {
    if (this.isTagGrid) {
      return this.renderTagGridBreadcrumbs();
    }

    if (this.isPDP) {
      return this.renderPDPBreadCrumbs();
    }

    return null;
  }

  render() {
    const { classes: { container, breadcrumbLink } } = this.props;

    return (
      <div className={container}>
        <Link route="/"><span className={breadcrumbLink}>Home</span></Link>
        {this.renderBreadcrumbs()}
      </div>
    );
  }
}

export default withTags(Breadcrumbs);
