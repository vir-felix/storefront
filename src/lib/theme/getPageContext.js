import { SheetsRegistry } from "jss";
import { createGenerateClassName } from "@material-ui/core/styles";
import theme from "./reactionTheme";

/**
 * @summary Creates and returns a page context object
 * @returns {Object} Page context object
 */
function createPageContext() {
  return {
    theme,
    // This is needed in order to deduplicate the injection of CSS in the page.
    sheetsManager: new Map(),
    // This is needed in order to inject the critical CSS.
    sheetsRegistry: new SheetsRegistry(),
    // The standard class name generator.
    generateClassName: createGenerateClassName()
  };
}

/**
 * Either creates a new or returns an existing page context object. If called from Node, creates
 * a new object every time. Otherwise creates and caches an object in the browser and
 * subsequent calls will return that.
 *
 * @summary Gets a page context object
 * @returns {Object} Page context object
 */
export default function getPageContext() {
  // Make sure to create a new context for every server-side request so that data
  // isn't shared between connections (which would be bad).
  if (!process.browser) {
    return createPageContext();
  }

  // Reuse context on the client-side.
  if (!global.__INIT_MATERIAL_UI__) {
    global.__INIT_MATERIAL_UI__ = createPageContext();
  }

  return global.__INIT_MATERIAL_UI__;
}
