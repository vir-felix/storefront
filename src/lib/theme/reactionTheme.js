import { createMuiTheme } from "material-ui/styles";

const theme = createMuiTheme({
  palette: {
    primary: {
      light: "#26B0F9",
      main: "#1999DD",
      dark: "#172F3C",
      contrastText: "#FFFFFF",
      borderColor: "#CCCCCC"
    },
    secondary: {
      light: "#5d8ea9",
      main: "#5E7480",
      dark: "#1D1D1D",
      contrastText: "#000000"
    },
    error: {
      light: "#E54F5D",
      main: "#CD3F4C",
      dark: "#3C1F21",
      contrastText: "#FFFFFF"
    },
    reaction: {
      activeElementBorderColor: "#94E8D1",
      activeElementBackground: "#E6E6E6"
    }

  },
  typography: {
    fontFamily: "Source Sans Pro, Helvetica Neue, Helvetica, sans-serif",
    fontSize: 16,
    fontWeightLight: 200,
    fontWeightRegular: 400,
    fontWeightMedium: 700,
    fontWeightBold: 700
  }
});

export default theme;
