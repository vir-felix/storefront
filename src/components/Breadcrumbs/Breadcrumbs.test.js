import React from "react";
import renderer from "react-test-renderer";
import { MuiThemeProvider } from "@material-ui/core/styles";
import { Provider } from "mobx-react";
import theme from "custom/reactionTheme";
import Breadcrumbs from "./Breadcrumbs";

const tags = {
  edges: [
    {
      node: {
        _id: "v2hE7fzL6cJniwgSm",
        name: "Tag A",
        slug: "tag-a",
        shopId: "J8Bhq3uTtdgwZx3rz",
        isTopLevel: true,
        updatedAt: "2018-06-11T18:01:09.568Z",
        createdAt: "2018-06-11T18:00:48.414Z",
        isDeleted: false,
        isVisible: true,
        groups: [],
        subTagIds: [
          "s3nybDz4QPkmapETc",
          "JB4FRBiWduNsxhhhn",
          "gRM3ADcY77eNJ7Brm"
        ]
      }
    }, {
      node: {
        _id: "s3nybDz4QPkmapETc",
        name: "Tag A-1",
        slug: "tag-a-1",
        shopId: "J8Bhq3uTtdgwZx3rz",
        isTopLevel: false,
        updatedAt: "2018-06-11T18:01:02.411Z",
        createdAt: "2018-06-11T18:01:02.411Z",
        isDeleted: false,
        isVisible: true,
        groups: []
      }
    }, {
      node: {
        _id: "JB4FRBiWduNsxhhhn",
        name: "Tag A-2",
        slug: "tag-a-2",
        shopId: "J8Bhq3uTtdgwZx3rz",
        isTopLevel: false,
        updatedAt: "2018-06-11T18:01:06.975Z",
        createdAt: "2018-06-11T18:01:06.975Z",
        isDeleted: false,
        isVisible: true,
        groups: []
      }
    }, {
      node: {
        _id: "gRM3ADcY77eNJ7Brm",
        name: "Tag A-3",
        slug: "tag-a-3",
        shopId: "J8Bhq3uTtdgwZx3rz",
        isTopLevel: false,
        updatedAt: "2018-06-11T18:01:09.565Z",
        createdAt: "2018-06-11T18:01:09.565Z",
        isDeleted: false,
        isVisible: true,
        groups: []
      }
    }
  ]
};

const routingStore = {
  tag: {
    _id: "v2hE7fzL6cJniwgSm",
    name: "Tag A",
    slug: "tag-a",
    shopId: "J8Bhq3uTtdgwZx3rz",
    isTopLevel: true,
    updatedAt: "2018-06-11T18:01:09.568Z",
    createdAt: "2018-06-11T18:00:48.414Z",
    isDeleted: false,
    isVisible: true,
    groups: [],
    relatedTagIds: [
      "s3nybDz4QPkmapETc",
      "JB4FRBiWduNsxhhhn",
      "gRM3ADcY77eNJ7Brm"
    ]
  }
};

test("tagGrid top level tag", () => {
  const nodes = tags.edges.map((edge) => edge.node);
  const tag = nodes.find((findTag) => findTag._id === "v2hE7fzL6cJniwgSm");

  const component = renderer.create((
    <MuiThemeProvider theme={theme}>
      <Provider primaryShopId={"J8Bhq3uTtdgwZx3rz"} routingStore={routingStore}>
        <Breadcrumbs isTagGrid={true} tag={tag} tags={tags} />
      </Provider>
    </MuiThemeProvider>
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("tagGrid secondlevel tag", () => {
  const nodes = tags.edges.map((edge) => edge.node);
  const tag = nodes.find((findTag) => findTag._id === "s3nybDz4QPkmapETc");

  const component = renderer.create((
    <MuiThemeProvider theme={theme}>
      <Provider routingStore={routingStore}>
        <Breadcrumbs isTagGrid={true} tag={tag} tags={tags} />
      </Provider>
    </MuiThemeProvider>
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
