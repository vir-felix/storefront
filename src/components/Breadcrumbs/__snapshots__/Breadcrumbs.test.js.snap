// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tagGrid secondlevel tag 1`] = `
<div
  className="Breadcrumbs-container-1"
>
  <a
    className="WithTracking-Link--link-4"
    href="/"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="link"
    tabIndex={0}
    tracking={
      Object {
        "getTrackingData": [Function],
        "trackEvent": [Function],
      }
    }
  >
    <span
      className="Breadcrumbs-breadcrumbLink-2"
    >
      Home
    </span>
  </a>
  <svg
    aria-hidden="true"
    className="MuiSvgIcon-root-5 Breadcrumbs-breadcrumbIcon-3"
    color={undefined}
    focusable="false"
    viewBox="0 0 24 24"
  >
    <g>
      <path
        d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
      />
    </g>
  </svg>
  <a
    className="WithTracking-Link--link-4"
    href="/tag/tag-a"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="link"
    tabIndex={0}
    tracking={
      Object {
        "getTrackingData": [Function],
        "trackEvent": [Function],
      }
    }
  >
    <span
      className="Breadcrumbs-breadcrumbLink-2"
    >
      Tag A
    </span>
  </a>
  <svg
    aria-hidden="true"
    className="MuiSvgIcon-root-5 Breadcrumbs-breadcrumbIcon-3"
    color={undefined}
    focusable="false"
    viewBox="0 0 24 24"
  >
    <g>
      <path
        d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
      />
    </g>
  </svg>
  <a
    className="WithTracking-Link--link-4"
    href="/tag/tag-a-1"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="link"
    tabIndex={0}
    tracking={
      Object {
        "getTrackingData": [Function],
        "trackEvent": [Function],
      }
    }
  >
    <span
      className="Breadcrumbs-breadcrumbLink-2"
    >
      Tag A-1
    </span>
  </a>
</div>
`;

exports[`tagGrid top level tag 1`] = `
<div
  className="Breadcrumbs-container-1"
>
  <a
    className="WithTracking-Link--link-4"
    href="/"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="link"
    tabIndex={0}
    tracking={
      Object {
        "getTrackingData": [Function],
        "trackEvent": [Function],
      }
    }
  >
    <span
      className="Breadcrumbs-breadcrumbLink-2"
    >
      Home
    </span>
  </a>
  <svg
    aria-hidden="true"
    className="MuiSvgIcon-root-5 Breadcrumbs-breadcrumbIcon-3"
    color={undefined}
    focusable="false"
    viewBox="0 0 24 24"
  >
    <g>
      <path
        d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
      />
    </g>
  </svg>
  <a
    className="WithTracking-Link--link-4"
    href="/tag/tag-a"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="link"
    tabIndex={0}
    tracking={
      Object {
        "getTrackingData": [Function],
        "trackEvent": [Function],
      }
    }
  >
    <span
      className="Breadcrumbs-breadcrumbLink-2"
    >
      Tag A
    </span>
  </a>
</div>
`;
